obj/generators.h

```diff
@@ -36,12 +44,13 @@ struct FakeRandomGen : public RandomGen {
 // and origin of the source position
 class PointSource final {
 public:
-  PointSource() = delete;
-  explicit PointSource(
-      float freq, int numOfRaysPerRow, float SampleSize, RandomGen *randomGen,
-      core::Vec3 origin = core::Vec3(0, 0, constants::kDefaultSimulationRadius))
+  PointSource() = default;
+  explicit PointSource(float freq, int numOfRaysPerRow, float SampleSize,
+                       RandomGen *randomGen, float simulationRadius,
+                       float sourcePower)
```

Comment:
Why is this called _power_ while this represents an energy. Do you specify the time interval somewhere so that power can be converted to energy? Or did I miss something?

If this is just energy for one ray, why not calling it `rayEnergy`? Also this will increase while your numOfRaysPerRow increase, to please decide if you want to provide total source energy or it is easier to provide per ray value. I am **guessing** it is easier to provide total.